### ISSUE TITLE
[META/Proposal] OpenSearch Project Should Adopt Apache Foundation's stance on the use of GenAI

### ISSUE THREAD

**Issue by scrawfor99** [opensearch-project/OpenSearch#11472]
Tl;DR: GenAI use is already rampant across many opensource projects websites. We (OpenSearch) should acknowledge the value of these tools but also enforce responsible use. I propose the adoption of the Apache Foundation's Generative AI Guidance (https://www.apache.org/legal/generative-tooling.html) or something similar. 

**Describe the solution you'd like**
We should add a new line to the sign-off that says you attest you reported any use of generative AI tooling. To report the use we can just add a new checkbox alongside the several we already have on pull requests. 

I think this solution gives the project leeway in addressing generative AI use on a case by case basis while also getting in front of some of the more problematic practices genAI has led to. There has been discussion around examples of PRs from various projects where documentation and/or code has been run through gen AI tooling and then PRs opened to make these changes. 

These contributions have been referred to as "low effort" or "low value" in context of those changes and demonstrated the need for some guidelines. 

OpenSearch is millions of lines of code and changes to improve our readability of efficiency are often useful. However, there should be expectations about how and when genAI use is appropriate. I think having this evident on PRs will make it easier for maintainers to be judicious in reviewing pull requests. 

**Describe alternatives you've considered**
Leaving things as is; banning genAI use; no restrictions whatsoever. 

**Additional context**
I can share example PRs of what I am referring to, but please reach out to me directly on the public slack as I would prefer not to publicly use anyone's work as a "bad" example.

**Comment by dblock** [comment]
@scrawfor99 

1. Maybe copy paste some anonymized/generic screenshots? 
2. I am moving this issue to .github. That's where any project-wide policy would go.
